cmd/ipfs/daemon.go

```diff
@@ -488,3 +503,16 @@ func merge(cs ...<-chan error) <-chan error {
 	}()
 	return out
 }
+
+func runGC(req cmds.Request, repo repo.Repo, node *core.IpfsNode) error {
+	disableGC, _, err := req.Option(disableGCKwd).Bool()
+	if err != nil {
+		return err
+	}
+	if disableGC {
+		return nil
+	}
+
+	ctx := req.Context()
+	return fsrepo.PeriodicGC(ctx, repo, node)
```

Comment:
`PeriodicGC`'s implementation is generic, but there is no way to write generic interface methods, therefore the only option is `repo.PeriodicGC(ctx, myrepo, node)` -- not as an interface method.

The reason for `PeriodicTasks` is so that there is only 1 for loop and tasks are to be added as new select cases? Will abstract this out when there is an additional repo periodic task in addition to repo gc (e.g. periodic version snapshot & periodic remove last-n snapshots?).

For option 3, this will cause running ephemeral node to be slightly more expensive due to the periodic tasks (but this issue alone isn't enough to get option 3 out of the picture).